gremlin-dotnet/src/Gremlin.Net/Driver/GremlinClient.cs

```diff
@@ -69,6 +69,21 @@ public class GremlinClient : IGremlinClient
             var writer = graphSONWriter ?? new GraphSON3Writer();
             var connectionFactory = new ConnectionFactory(gremlinServer, reader, writer, mimeType ?? DefaultMimeType,
                 webSocketConfiguration, sessionId);
+
+            // make sure one connection in pool as session mode
+            if (!String.IsNullOrEmpty(sessionId))
+            {
+                if (connectionPoolSettings != null)
+                {
+                    if (connectionPoolSettings.PoolSize != 1)
+                        throw new ArgumentOutOfRangeException(nameof(connectionPoolSettings), "Session Client PoolSize must be 1!");
```

Comment:
I am not sure whether every user will understand what `Session Client` means as we don't use that term any where else I think. Could you maybe reword it to something like: _Only a pool size of 1 is supported in session mode._?

Reply:
ok, session mode should be better